### DISCUSSION TITLE
Use MediatR and MS.DI library in Avalonia project

### DISCUSSION THREAD

**Discussion by rammfire** [AvaloniaUI/Avalonia#7465] [Q&A]
Hi all!
I need you help with explenation about how right configure DI services, inside which i can call MediatR method in my ViewModel class.
Now, in the Program.cs class, I have the following services configured:
```
 public static void Main(string[] args)
        {
            var services = new ServiceCollection()
                .AddDbContext<DataBaseContext>(options =>
                {
                    options.UseSqlite("Data Source=Players.db;");
                })
                .AddTransient<IDbContext, DataBaseContext>()
                .AddMediatR(typeof(InsertNewPlayerInDB.InsertNewPlayerInDBCommand).Assembly)
                .AddTransient<IMediatorServices, MediatorServices>()
                .BuildServiceProvider();

            BuildAvaloniaApp().StartWithClassicDesktopLifetime(args);
        }
```

And in the ViewModel I have the following code:
```
public class UserLoginOrRegistrationViewModel: ViewModelBase
    {
        public readonly MediatR.IMediator _mediator;

        public ReactiveCommand<Unit,Unit> Command { get ; }

        public async void StartCommand()
        {
            await _mediator.Send(new InsertNewPlayerInDBCommand());
        }
        public UserLoginOrRegistrationViewModel(MediatR.IMediator mediator)
        {
            _mediator = mediator;
            Command = ReactiveCommand.Create(StartCommand);
        }
    }
```
In fact, I get an error message saying that Framework cannot find an argument that matches a formal parameter in the constructor.

**Comment by rammfire** [comment]
I still need help because I didn't find a solution

**Comment by maxkatz6** [comment]
Not sure why you need to use MediatR with GUI application, but it's okay. Also it doesn't seem to be related to avalonia at all. 
Next, you didn't provide specific exception message and stacktrace. We don't see where you want to resolve the view model. 
It's not much what we actually can help here right now.

**Reply by rammfire** [reply]
In short, this is part of my educational process for DI. Now I will write more details of what needs to be done.
